.github/workflows/issues.yml

```diff
@@ -1,29 +1,27 @@
 ---
   name: Create JIRA ticket for new issues
   
+  # Creates and updates jira tickets that sync with GitHub Issues events.
   on:
     issues:
-      types: [opened]
-      
-  permissions: 
-    issues: write
-    contents: read
+      types: [opened, reopened, closed]
   jobs:
     jira_task:
       name: Create Jira issue
+      if: github.event.action == 'opened'
       runs-on: ubuntu-latest
+      env:
+        GITHUB_TOKEN: ${{ secrets.APIX_BOT_PAT }}
```

Comment:
Why we need github token here?

Reply:
Awesome

Reply:
I think you need it: GH Action cannot opens/close/update other issue/PR opened or created by another GH action. in order to do so, you need a personal access token. This is a limitation of GH action to prevent to have an infinite loop 

see https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow

> When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN, with the exception of workflow_dispatch and repository_dispatch, will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs. For example, if a workflow run pushes code using the repository's GITHUB_TOKEN, a new workflow will not run even when the repository contains a workflow configured to run when push events occur.

Reply:
@andreaangiolillo good catch, thanks, will put them back